From 98b692708f63472c7639abddc724100c1d190b06 Mon Sep 17 00:00:00 2001
From: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Date: Tue, 17 Aug 2021 12:21:23 +0300
Subject: [PATCH 19/19] s32gen1: pcie: Fix some static code issues

commit 98b692708f63472c7639abddc724100c1d190b06 from
https://source.codeaurora.org/external/autobsps32/linux

1. Add 'static' keyword to members used only in this file.
2. Fix some incorrect casts.

Upstream-Status: Pending

Issue: ALB-7518
Signed-off-by: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/pci/controller/dwc/pci-s32gen1.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-s32gen1.c b/drivers/pci/controller/dwc/pci-s32gen1.c
index 94fdd5a2c289..e95ba7a9be16 100644
--- a/drivers/pci/controller/dwc/pci-s32gen1.c
+++ b/drivers/pci/controller/dwc/pci-s32gen1.c
@@ -151,7 +151,7 @@
 			BAR_ ## bar_no, \
 			PCIE_EP_BAR ## bar_no ## _INIT}
 
-struct pci_epf_bar s32gen1_ep_bars[] = {
+static struct pci_epf_bar s32gen1_ep_bars[] = {
 		PCIE_EP_BAR_INIT(0),
 		PCIE_EP_BAR_INIT(1),
 		PCIE_EP_BAR_INIT(2),
@@ -159,7 +159,8 @@ struct pci_epf_bar s32gen1_ep_bars[] = {
 		PCIE_EP_BAR_INIT(4),
 		PCIE_EP_BAR_INIT(5)
 };
-int s32gen1_ep_bars_en[] = {
+
+static int s32gen1_ep_bars_en[] = {
 		PCIE_EP_BAR0_EN_DIS,
 		PCIE_EP_BAR1_EN_DIS,
 		PCIE_EP_BAR2_EN_DIS,
@@ -853,28 +854,28 @@ static int s32gen1_pcie_dt_init(struct platform_device *pdev,
 	if (IS_ERR(pcie->dbi_base))
 		return PTR_ERR(pcie->dbi_base);
 	dev_dbg(dev, "dbi: %pR\n", res);
-	dev_dbg(dev, "dbi virt: 0x%llx\n", (u64)pcie->dbi_base);
+	dev_dbg(dev, "dbi virt: 0x%p\n", pcie->dbi_base);
 
 	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "dbi2");
 	pcie->dbi_base2 = devm_ioremap_resource(dev, res);
 	if (IS_ERR(pcie->dbi_base2))
 		return PTR_ERR(pcie->dbi_base2);
 	dev_dbg(dev, "dbi2: %pR\n", res);
-	dev_dbg(dev, "dbi2 virt: 0x%llx\n", (u64)pcie->dbi_base2);
+	dev_dbg(dev, "dbi2 virt: 0x%p\n", pcie->dbi_base2);
 
 	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "atu");
 	dev_dbg(dev, "atu: %pR\n", res);
 	pcie->atu_base = devm_ioremap_resource(dev, res);
 	if (IS_ERR(pcie->atu_base))
 		return PTR_ERR(pcie->atu_base);
-	dev_dbg(dev, "atu virt: 0x%llx\n", (u64)pcie->atu_base);
+	dev_dbg(dev, "atu virt: 0x%p\n", pcie->atu_base);
 
 	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "ctrl");
 	s32_pp->ctrl_base = devm_ioremap_resource(dev, res);
 	if (IS_ERR(s32_pp->ctrl_base))
 		return PTR_ERR(s32_pp->ctrl_base);
 	dev_dbg(dev, "ctrl: %pR\n", res);
-	dev_dbg(dev, "ctrl virt: 0x%llx\n", (u64)s32_pp->ctrl_base);
+	dev_dbg(dev, "ctrl virt: 0x%p\n", s32_pp->ctrl_base);
 
 	s32_pp->linkspeed = of_pci_get_max_link_speed(np);
 	if (s32_pp->linkspeed < GEN1 || s32_pp->linkspeed > GEN3) {
-- 
2.17.1

